app/src/main/java/org/jtaccuino/app/studio/SheetManagerSPIImpl.java

```diff
@@ -15,14 +15,51 @@
  */
 package org.jtaccuino.app.studio;
 
+import java.nio.file.Path;
+import javafx.beans.property.ReadOnlyListWrapper;
+import javafx.beans.property.SimpleListProperty;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
 import org.jtaccuino.app.common.NotebookPersistence;
+import org.jtaccuino.app.persistence.FilePersistence;
 import org.jtaccuino.core.ui.Sheet;
+import org.jtaccuino.core.ui.api.SheetManager;
 import org.jtaccuino.core.ui.spi.SheetManagerSPI;
 
 public class SheetManagerSPIImpl extends SheetManagerSPI{
 
+    private ObservableList<SheetManager.RecentFile> recentFiles =
+            FXCollections.observableArrayList(
+                    FilePersistence.getDefault().recentFiles.stream().map(rf -> from(rf.path())).toList());
+    private ReadOnlyListWrapper<SheetManager.RecentFile> unmodifiableRecentFiles = new ReadOnlyListWrapper<>(recentFiles);
+
+    private static SheetManager.RecentFile from(Path path) {
+        return new SheetManager.RecentFile(path.getName(path.getNameCount() - 1).toString(), path);
```

Comment:
What does this do?
Do you want the file name as the displayName (first param)?
If so, why not use


```suggestion
        return new SheetManager.RecentFile(path.getFileName().toString(), path);
```

Reply:
Will check.